pkg/istio-agent/xds_proxy.go

```diff
@@ -485,8 +479,8 @@ func (p *XdsProxy) handleUpstreamRequest(con *ProxyConnection) {
 				p.ecdsLastNonce.Store(req.ResponseNonce)
 			}
 			if err := sendUpstream(con.upstream, req); err != nil {
-				err = fmt.Errorf("upstream [%d] send error for type url %s: %v", con.conID, req.TypeUrl, err)
-				con.upstreamError <- err
```

Comment:
since requestsChan is unbounded channel, it can buffer many requests, but once the upstream connection closed,  no thread is receiving from `upstreamError`, this lead to goroutine leak

Reply:
yeah with this, we donot need cahnnel capacity. it will not block as the receiver is there

Reply:
As to why it does not block, please see https://github.com/istio/istio/blob/89b9f5b574e2aad0149c0b032fa7b68c17461974/pkg/istio-agent/xds_proxy.go#L398-L421

in `handleUpstream`, we always recv these two error channels, if one err received,  this function exit, and unregisterStream called

https://github.com/istio/istio/blame/89b9f5b574e2aad0149c0b032fa7b68c17461974/pkg/istio-agent/xds_proxy.go#L332, and stopChan is closed.

So next time an error happen on the same connection won't block